From 6ce0da745857fb7c3cf94a63667662a440688367 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Mon, 3 Jun 2024 18:55:19 +0200
Subject: [PATCH 0944/1695] dts: st: add low-power timer 2-5 nodes on
 stm32mp211

Add low-power timer (LPTimer) 2/3/4/5 support on STM32MP21 SoC.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 301a07a1008de3bb600d5328ecbce7e1616c2f15 dts: st: add low-power timer 2-5 nodes on stm32mp211]
Change-Id: I6c4090b2419e80274ced979c268c95b6886e53ea
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/377298
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 132 ++++++++++++++++++++++++++++++
 1 file changed, 132 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 8ae768b32..74b8028c5 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -255,6 +255,39 @@
 				};
 			};
 
+			lptimer2: timer@400a0000 {
+				compatible = "st,stm32mp21-lptimer";
+				reg = <0x400a0000 0x400>;
+				interrupts = <GIC_SPI 180 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_LPTIM2>;
+				clock-names = "mux";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "disabled";
+
+				counter {
+					compatible = "st,stm32mp21-lptimer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32mp21-pwm-lp";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer {
+					compatible = "st,stm32mp21-lptimer-timer";
+					status = "disabled";
+				};
+
+				trigger@1 {
+					compatible = "st,stm32mp21-lptimer-trigger";
+					reg = <1>;
+					status = "disabled";
+				};
+			};
+
 			usart2: serial@400e0000 {
 				compatible = "st,stm32h7-uart";
 				reg = <0x400e0000 0x400>;
@@ -392,6 +425,105 @@
 				access-controllers = <&rifsc STM32MP21_RIFSC_I2C3_ID>;
 				status = "disabled";
 			};
+
+			lptimer3: timer@46050000 {
+				compatible = "st,stm32mp21-lptimer";
+				reg = <0x46050000 0x400>;
+				interrupts = <GIC_SPI 181 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_LPTIM3>;
+				clock-names = "mux";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "disabled";
+
+				counter {
+					compatible = "st,stm32mp21-lptimer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32mp21-pwm-lp";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer {
+					compatible = "st,stm32mp21-lptimer-timer";
+					status = "disabled";
+				};
+
+				trigger@2 {
+					compatible = "st,stm32mp21-lptimer-trigger";
+					reg = <2>;
+					status = "disabled";
+				};
+			};
+
+			lptimer4: timer@46060000 {
+				compatible = "st,stm32mp21-lptimer";
+				reg = <0x46060000 0x400>;
+				interrupts = <GIC_SPI 182 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_LPTIM4>;
+				clock-names = "mux";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "disabled";
+
+				counter {
+					compatible = "st,stm32mp21-lptimer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32mp21-pwm-lp";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer {
+					compatible = "st,stm32mp21-lptimer-timer";
+					status = "disabled";
+				};
+
+				trigger@3 {
+					compatible = "st,stm32mp21-lptimer-trigger";
+					reg = <3>;
+					status = "disabled";
+				};
+			};
+
+			lptimer5: timer@46070000 {
+				compatible = "st,stm32mp21-lptimer";
+				reg = <0x46070000 0x400>;
+				interrupts = <GIC_SPI 183 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_KER_LPTIM5>;
+				clock-names = "mux";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "disabled";
+
+				counter {
+					compatible = "st,stm32mp21-lptimer-counter";
+					status = "disabled";
+				};
+
+				pwm {
+					compatible = "st,stm32mp21-pwm-lp";
+					#pwm-cells = <3>;
+					status = "disabled";
+				};
+
+				timer {
+					compatible = "st,stm32mp21-lptimer-timer";
+					status = "disabled";
+				};
+
+				trigger@4 {
+					compatible = "st,stm32mp21-lptimer-trigger";
+					reg = <4>;
+					status = "disabled";
+				};
+			};
 		};
 
 		iac: iac@42090000 {
-- 
2.47.2

